src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -655,60 +647,101 @@ void ClusterQueueHelper::onQueueContextAssigned(
         const ClusterStatePartitionInfo& pinfo = d_clusterState_p->partition(
             pid);
 
-        logMsg << "Queue '" << queueContext->uri()
-               << "' now assigned to Partition [" << pid << "]";
-        if (pinfo.primaryNode()) {
-            logMsg << " (" << pinfo.primaryNode()->nodeDescription() << ").";
-        }
-        else {
-            logMsg << " (*no primary*).";
-        }
-
-        if (queueContext->d_liveQInfo.d_pending.empty()) {
-            // If no pending contexts, nothing more to do here but log.
-            logMsg << " No pending contexts for the queue.";
-            BALL_LOG_INFO << logMsg.str();
-            return;  // RETURN
-        }
-
         if (!d_cluster_p->isFSMWorkflow() &&
             bmqp_ctrlmsg::NodeStatus::E_AVAILABLE !=
                 d_clusterData_p->membership().selfNodeStatus()) {
             // Self is not available, we have to postpone processing the queue
             // opening.
 
-            logMsg << " There are "
-                   << queueContext->d_liveQInfo.d_pending.size()
-                   << " associated pending contexts, but self is not "
-                   << "AVAILABLE. Current self node status: "
-                   << d_clusterData_p->membership().selfNodeStatus();
-            BALL_LOG_INFO << logMsg.str();
-            return;  // RETURN
+            isAvailable = false;
         }
-
-        if (!pinfo.primaryNode() ||
-            bmqp_ctrlmsg::PrimaryStatus::E_ACTIVE != pinfo.primaryStatus()) {
+        else if (!pinfo.primaryNode() ||
+                 bmqp_ctrlmsg::PrimaryStatus::E_ACTIVE !=
+                     pinfo.primaryStatus()) {
             // We don't have a primary or primary is passive, we have to
             // postpone processing the queue opening.
 
-            logMsg << " There are "
-                   << queueContext->d_liveQInfo.d_pending.size()
-                   << " associated pending contexts, waiting for an ACTIVE "
-                   << "primary for the partition. Current primary: "
-                   << (pinfo.primaryNode()
-                           ? pinfo.primaryNode()->nodeDescription()
-                           : "** null **");
-
-            BALL_LOG_INFO << logMsg.str();
-            return;  // RETURN
+            haveActivePrimary = false;
         }
     }
 
-    logMsg << " Proceeding with " << queueContext->d_liveQInfo.d_pending.size()
-           << " associated pending contexts.";
-    BALL_LOG_INFO << logMsg.str();
+    BALL_LOGTHROTTLE_INFO_BLOCK(k_MAX_INSTANT_MESSAGES, k_NS_PER_MESSAGE)
+    {
+        bmqu::MemOutStream logMsg(d_allocator_p);
+        logMsg << d_cluster_p->description() << ": ";
+
+        if (d_cluster_p->isRemote()) {
+            BSLS_ASSERT_SAFE(mqbs::DataStore::k_INVALID_PARTITION_ID == pid);
```

Comment:
Have a comment about this assert above, it's better to move it from here to its original place